client/index.jsx

```diff
@@ -1,8 +1,51 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import App from './app';
+import React, { useRef } from 'react';
+import reactDOM from 'react-dom/client';
 
 const container = document.querySelector('#root');
-const root = ReactDOM.createRoot(container);
+const root = reactDOM.createRoot(container);
 
-root.render(<App />);
+function MusicWidget() {
+  const widgetRef = useRef(null);
+  return (
+    <div
+      className="music-widget"
+      ref={widgetRef}
+      style={{ width: '400px', height: '200px', background: '#93B1A7' }}
+      onMouseDown={e => {
+        e.preventDefault();
+        widgetRef.current.style.position = 'absolute';
+        const shiftX = e.clientX - widgetRef.current.getBoundingClientRect().left;
+        const shiftY = e.clientY - widgetRef.current.getBoundingClientRect().top;
+
+        const onMouseMove = e => {
+          widgetRef.current.style.left = e.pageX - shiftX + 'px';
+          widgetRef.current.style.top = e.pageY - shiftY + 'px';
+        };
+
+        document.addEventListener('mousemove', onMouseMove);
+
+        document.addEventListener('mouseup', () => {
```

Comment:
I see. When the mouse is clicked you register a mousemove listener and when they release the mouse you remove it. Thanks for researching this and figuring it out! I've now learned something.